storage.go

```diff
@@ -241,22 +232,14 @@ func (s *Storage) List(ctx context.Context, k microstorage.K) ([]microstorage.KV
 }
 
 func (s *Storage) Put(ctx context.Context, kv microstorage.KV) error {
-	var err error
-
-	var body []byte
-	{
-		v := storageConfigJSONPatch{}
-		v.Spec.Storage.Data = map[string]*string{}
-		value := kv.Val()
-		v.Spec.Storage.Data[kv.Key()] = &value
-
-		body, err = json.Marshal(v)
-		if err != nil {
-			return microerror.Mask(err)
-		}
+	storageConfig, err := s.g8sClient.CoreV1alpha1().StorageConfigs(s.namespace.Name).Get(s.name, apismetav1.GetOptions{})
+	if err != nil {
+		return microerror.Mask(err)
 	}
 
-	_, err = s.g8sClient.CoreV1alpha1().StorageConfigs(s.namespace.Name).Patch(s.name, types.MergePatchType, body)
+	storageConfig.Spec.Storage.Data[kv.Key()] = kv.Val()
+
+	_, err = s.g8sClient.CoreV1alpha1().StorageConfigs(s.namespace.Name).Update(storageConfig)
```

Comment:
Are you sure it works? I remember @corest having issues with resource version IIRC.

Reply:
The integration test says it works. I was thinking if we want to have retries in here because in theory it can happen that the object gets updated meanwhile and then we fail. WDYT?

Reply:
Not sure. But in that case https://github.com/giantswarm/microstorage/tree/master/retrystorage can be used.

Reply:
But this means we have to use it everywhere we want to retry explicitly right? At least we have to configure it once at runtime. We should do this everywhere anyway. Cool, I go with this then.

Reply:
Yes, once at runtime and you are done. Defaults are there. I'd say it's good enough and would be weird to retry inside a retry.